python/source/flight.rst

```diff
@@ -605,3 +605,102 @@ Or if we use the wrong credentials on login, we also get an error:
+
+Since `dotnet dev-certs` does not let you export Public and Private keys directly we need to convert the .pfx file. 
+There are several way to achieve this and this tutorial uses OpenSSL (using Windows Subsystem for Linux) 
+to perform the conversion as per this `IBM article`_.
+
+**Step 3 - Running a server with tls enabled**
+
+We're going to use the pyarrow server example available on the `GitHub repo`_. To run the server with TLS enabled, the python script should be 
+called with the path to the public and private keys.
+.. testcode::
+    python server.py --tls CERTFILE <PathToPublicCertificate> --tls KEYFILE <PathToPrivateKey>
+
+Assuming the path was valid, you should see ``Serving on grpc+tls://localhost:5005``. The server is now being served on a port set in the code (or by you).
+
+**Step 4 - Securely Connecting a client to the Server**
+Suppose we want to connect to the client and push some data to it. The following code securely sends information to the server using TLS encryption. 
+There is also the option to use mutual TLS encryption using both the public and private key, but we will assume the client will likely only have 
+the public certificate.
+.. testcode::
+    import pyarrow
+    import pyarrow.flight
+    import pandas as pd
+    
+    # Assumes incoming data object is a Dataframe
+    def pushToServer(name, data, client):
+        objectToSend = pyarrow.Table.from_pandas(data)
+        writer, _ = client.do_put(pyarrow.flight.FlightDescriptor.for_path(name), objectToSend.schema)
+        writer.write_table(objectToSend)
+        writer.close()
+    
+    def getClient():
+        
+        return pyarrow.flight.FlightClient("grpc+tcp://localhost:5005")
+    
+    def _add_common_arguments(parser):
+        parser.add_argument('--tls', action='store_true',
+                            help='Enable transport-level security')
+        parser.add_argument('--tls-roots', default=None,
+                            help='Path to trusted TLS certificate(s)')
+        parser.add_argument("--mtls", nargs=2, default=None,
+                            metavar=('CERTFILE', 'KEYFILE'),
+                            help="Enable transport-level security")
+                            
+    def main():
+        parser = argparse.ArgumentParser()
+        args = parser.parse_args()
+        connection_args = {}
+        scheme = "grpc+tls"
+        
+        if args.tls:
+            
+            if args.tls_roots:
+                with open(args.tls_roots, "rb") as root_certs:
+                    connection_args["tls_root_certs"] = root_certs.read()
+        if args.mtls:
+            with open(args.mtls[0], "rb") as cert_file:
+                tls_cert_chain = cert_file.read()
+            with open(args.mtls[1], "rb") as key_file:
+                tls_private_key = key_file.read()
+            connection_args["cert_chain"] = tls_cert_chain
+            connection_args["private_key"] = tls_private_key
```

Comment:
IMO, we can split mutual TLS into a separate example to reduce the number of things going on here.

Reply:
(As in, we can tackle that in a separate PR - not asking you to double the workload here :slightly_smiling_face:)

Reply:
done :)